project/profiles/capacity4more/modules/c4m/restful/c4m_restful_quick_post/c4m_restful_quick_post.module

```diff
@@ -271,8 +272,9 @@ function c4m_restful_quick_post_form($group_id) {
     }
   }
 
-  // If all resources are turned off, disable the QP form.
-  if (empty($show_resources)) {
+  // If all resources are turned off, or user is not allowed to create a
+  // discussion, disable the QP form.
+  if (empty($show_resources) || !node_access('create', 'discussion')) {
```

Comment:
@anvmn 
this one also seems to be too generic,
did you test this for SA, GO, GA and GM separately ?